scio-smb/src/test/scala/com/spotify/scio/smb/SmbIOTest.scala

```diff
@@ -179,4 +192,106 @@ class SmbIOTest extends PipelineSpec {
       .run()
   }
 
+  "SortedBucketTap" should "work with SMB writes" in {
+    val tempFolder = Files.createTempDirectory("smb-tap").toFile
+    tempFolder.deleteOnExit()
+
+    val sc = ScioContext()
+    val tap = sc
+      .parallelize(accountsIterable)
+      .saveAsSortedBucket(
+        AvroSortedBucketIO
+          .write(classOf[String], "name", classOf[Account])
+          .to(tempFolder.getAbsolutePath)
+          .withNumBuckets(4)
+          .withNumShards(2)
+          .withFilenamePrefix("custom-prefix")
+      )
+
+    val result = sc.run().waitUntilDone()
```

Comment:
```suggestion
    val result = runWithRealContext() { sc =>
      sc.parallelize(accountsIterable)
         .saveAsSortedBucket(
        AvroSortedBucketIO
          .write(classOf[String], "name", classOf[Account])
          .to(tempFolder.getAbsolutePath)
          .withNumBuckets(4)
          .withNumShards(2)
          .withFilenamePrefix("custom-prefix")
    }.waitUntilDone()
```

Reply:
I don't know if this would work because we need to preserve `tap` (the result of `.saveAsSortedBucket`), but `runWithRealContext` only returns a `ScioExecutionContext`

Reply:
Maybe we can create a new testing helper method that returns a ClosedTap? like:

```scala
def runWithRealContext[T](
    options: PipelineOptions = PipelineOptionsFactory.create()
)(fn: ScioContext => Tap[T]): ClosedTap = {
    val sc = ScioContext(options)
    val tap = fn(sc)
    val result = sc.run().waitUntilDone()
    tap.get(result)
}

Reply:
Ha missed that too. Let's stick with current structure